Add SOP on how to update the observability operator CR manually

machi1990 opened the pull request:
/cc @davidffrench @pb82 @david-martin

pb82 commented on sops/update_observability_operator_cr.asciidoc:
````diff
@@ -0,0 +1,104 @@
+```
+
+You can change the Observability kind resource to update the value of the existing contents or add new fields. 
+The content of the CR has to match the https://github.com/bf2fc6cc711aee1a0c2a/observability-operator/blob/c6e8d9be4de1bec6d597ef7abfe8131d05bd5b36/config/crd/bases/observability.redhat.com_observabilities.yaml[Observability operator CRD]
````
the link should be inside parentheses to be rendered as an URL:

)https://github.com/bf2fc6cc711aee1a0c2a/observability-operator/blob/c6e8d9be4de1bec6d597ef7abfe8131d05bd5b36/config/crd/bases/observability.redhat.com_observabilities.yaml)[Observability operator CRD]

pb82 commented on sops/update_observability_operator_cr.asciidoc:
````diff
@@ -0,0 +1,104 @@
+Once the modifications of the CR are done, save the file and execute the following command to update the syncset.
+
+```
+jq 'del(.id)' cr.json | ocm patch /api/clusters_mgmt/v1/clusters/<cluster-id>/external_configuration/syncsets/ext-managed-application-services-observability
````
what's the `jq 'del(.id)'` for?

machi1990 replied:
This remove the id from the `cr.json` content. This is because the content of this cr are saved from the get command above.

pb82 reviewed (approved):
LGTM

david-martin commented:
Thanks @machi1990 @pb82.
I'll give this a look as well

david-martin commented on sops/update_observability_operator_cr.asciidoc:
````diff
@@ -0,0 +1,104 @@
+}
+```
+
+You can change the Observability kind resource to update the value of the existing contents or add new fields. 
````
Can this step be made easier?
The link is to the CRD, which is quite verbose and more likely to result in human error when parsing it.
Also, what values should be updated, new fields added and existing values left as is specifically?

machi1990 replied:
What's need to be updated is the content of the `cr.json` file. The CRD is only linked to see the schema. 
The values to be updated depends on what we want to update - the previous two cases it has been secrets but I did not want to call any specific fields as all fields can be updated.

david-martin replied:
Hmm, OK.
So the fields (and values) to update would be communicated as needed whenever this SOP has to be followed?

machi1990 replied:
> So the fields (and values) to update would be communicated as needed whenever this SOP has to be followed?

Yes. WDYT?  Do you want this to be called out in the SOP?

david-martin replied:
Good idea. Maybe something like `The fields and values to be modified should be communicated from the person or team requesting this SOP to be followed` ?

That would hint at the SOP potentially being needed many times

david-martin commented on sops/update_observability_operator_cr.asciidoc:
```diff
@@ -0,0 +1,104 @@
+
+== Prerequisites
+
+1. The target OSD cluster exists. Note the `cluster-id`.
```
What is the target OSD cluster?
Any/all clusters where the 'data plane' is running?

(I realise there's only 1 data plane cluster at present, but this SOP could be around for a while until https://issues.redhat.com/browse/MGDSTRM-882 is done, so I'd like to make it as clear as possible)

david-martin commented on sops/update_observability_operator_cr.asciidoc:
```diff
@@ -0,0 +1,104 @@
+== Prerequisites
+
+1. The target OSD cluster exists. Note the `cluster-id`.
+2. The Observability operator has been installed on the target cluster via the syncset approach. The id of the syncset could be `ext-managed-application-services-observability` (Prior to Jan, 2021) or `ext-managedservice-cluster-mgr` (as of Jan, 2021) depending on when the cluster was terraformed.
```
How to verify if the operator was installed this way (via syncset)?

machi1990 replied:
This will be via the `get` command given in the execution step.

david-martin commented on sops/update_observability_operator_cr.asciidoc:
```diff
@@ -0,0 +1,104 @@
+
+1. The target OSD cluster exists. Note the `cluster-id`.
+2. The Observability operator has been installed on the target cluster via the syncset approach. The id of the syncset could be `ext-managed-application-services-observability` (Prior to Jan, 2021) or `ext-managedservice-cluster-mgr` (as of Jan, 2021) depending on when the cluster was terraformed.
+The commands in this SOP assumes that the id is `ext-managed-application-services-observability`.
```
If the id is not what's expected, should it be updated to match?
Is it a problem if it doesn't match?

machi1990 replied:
The id could be one or the other. The current id is `ext-managedservice-cluster-mgr` but this only applies to new clusters, but the stage cluster is still using the old id `ext-managed-application-services-observability`.
The syncset will always have one or the other.

david-martin commented on sops/update_observability_operator_cr.asciidoc:
```diff
@@ -0,0 +1,104 @@
+2. The Observability operator has been installed on the target cluster via the syncset approach. The id of the syncset could be `ext-managed-application-services-observability` (Prior to Jan, 2021) or `ext-managedservice-cluster-mgr` (as of Jan, 2021) depending on when the cluster was terraformed.
+The commands in this SOP assumes that the id is `ext-managed-application-services-observability`.
+3. The https://github.com/openshift-online/ocm-cli[`ocm`] CLI.
+4. Cluster credentials are held that allow execution of `ocm` commands. 
```
There's a lot of different permissions that can be applied to OCM accounts.
What permissions or roles are needed specifically here?

david-martin commented on sops/update_observability_operator_cr.asciidoc:
````diff
@@ -0,0 +1,104 @@
+2. Execute the following to retrieve the content of the existing operator CR:
+
+```
+ocm get /api/clusters_mgmt/v1/clusters/<cluster-id>/external_configuration/syncsets/ext-managed-application-services-observability > cr.json
````
My understanding is a little lacking here.
I thought syncsets were created in the same cluster as where the hive operator is running.
Does this API do that for you?

davidffrench replied:
that is correct @david-martin . What is referred to as the trusted datapath are the syncset endpoints exposed through OCM cluster service which give the ability to interact with SyncSets on Hive.

david-martin commented on sops/update_observability_operator_cr.asciidoc:
````diff
@@ -0,0 +1,104 @@
+ocm get /api/clusters_mgmt/v1/clusters/<cluster-id>/external_configuration/syncsets/ext-managed-application-services-observability
+```
+
+2. Login in to the cluster and verify that the `managed-application-services-observability` 
````
Login to which cluster?
The OSD cluster?

machi1990 replied:
The OSD cluster.

I've added this clarification.

david-martin commented on sops/update_observability_operator_cr.asciidoc:
```diff
@@ -0,0 +1,104 @@
+
+== Description
+
+The purpose of this SOP is to update Observability operator CR on an OSD cluster after the initial installation of the operator was done.  
```
Can we add something here that explains why this has to be done manually?
Currently it reads like this has to always be done after the initial installation of the operator (which I believe isn't the case. Just for currently existing clusters or whenever the CR changes again in future and needs to be updated again in existing clusters)

davidffrench replied:
```suggestion
The purpose of this SOP is to update an existing observability operator installation (OLM Bundles, observability CR etc) on a data plane OSD cluster since the Managed Services API does not reconcile changes for existing installations.  
```

david-martin reviewed (changes requested):
See inline comments.

davidffrench commented on sops/update_observability_operator_cr.asciidoc:
```diff
@@ -0,0 +1,104 @@
+== Description
+
+The purpose of this SOP is to update Observability operator CR on an OSD cluster after the initial installation of the operator was done.  
+This is done to update the configuration of the operator e.g to update secrets of the alert manager.  
```
```suggestion
```

davidffrench commented on sops/update_observability_operator_cr.asciidoc:
```diff
@@ -0,0 +1,104 @@
+2. The Observability operator has been installed on the target cluster via the syncset approach. The id of the syncset could be `ext-managed-application-services-observability` (Prior to Jan, 2021) or `ext-managedservice-cluster-mgr` (as of Jan, 2021) depending on when the cluster was terraformed.
+The commands in this SOP assumes that the id is `ext-managed-application-services-observability`.
+3. The https://github.com/openshift-online/ocm-cli[`ocm`] CLI.
+4. Cluster credentials are held that allow execution of `ocm` commands. 
```
The individual running this SOP would also need permissions to see the OSD clusters created in the organisation used by the Managed Services API.

machi1990 replied:
Added.

davidffrench commented on sops/update_observability_operator_cr.asciidoc:
````diff
@@ -0,0 +1,104 @@
+Once the modifications of the CR are done, save the file and execute the following command to update the syncset.
+
+```
+jq 'del(.id)' cr.json | ocm patch /api/clusters_mgmt/v1/clusters/<cluster-id>/external_configuration/syncsets/ext-managed-application-services-observability
````
```suggestion
jq 'del(.id)' cr.json | ocm patch /api/clusters_mgmt/v1/clusters/<cluster-id>/external_configuration/syncsets/ext-managed-application-services-observability --body=cr.json
```

davidffrench commented on sops/update_observability_operator_cr.asciidoc:
```diff
@@ -0,0 +1,104 @@
+5. The https://stedolan.github.io/jq/[`jq`] CLI
+
+== Execute/Resolution
+
```
Would be good to set an env var for the cluster ID and SyncSet ID and just use them in the below commands. E.g.

`syncset-id=ext-managed-application-services-observability` and then just reference in the below commands as `$syncset-id`

machi1990 replied:
Thanks. I'll update this.

davidffrench reviewed (commented):
Nice SOP @machi1990 . Some small suggestions from me

machi1990 commented:
Thanks for the review @davidffrench @david-martin I am having a look now.

davidffrench reviewed (approved):
Thanks @machi1990 , this lgtm now

david-martin commented:
Taking another look. Thanks @machi1990 .

@robshelly can you take a look also as you'll likely be the person executing it if it happens this week

david-martin commented on sops/update_observability_operator_cr.asciidoc:
````diff
@@ -0,0 +1,106 @@
+
+```
+syncsetId=ext-managed-application-services-observability
+clusterId=target-cluster-id
````
Let's call out the target cluster id as a prereq,
and also include an example cmd on how to retrieve or determine this.

machi1990 replied:
Good point. I'll add it.

david-martin commented on sops/update_observability_operator_cr.asciidoc:
````diff
@@ -0,0 +1,106 @@
+```
+
+2. Login in to the OSD cluster and verify that the `managed-application-services-observability` 
+is there and the observability operator is running with the correct configuration. 
````
How to verify the operator is running with the correct configuration?
Is it different depending on the fields changed?
If so, are there a couple of examples that could be put here?

machi1990 replied:
I'll add a command to check the the observability CR has been updated to match the modified values.

david-martin reviewed (changes requested):
Thanks again @machi1990 
I appreciate the updates.

I've left a few more comments.
I know I'm being pedantic, but I really want to minimise the potential for human error.

machi1990 commented:
> Thanks again @machi1990
> I appreciate the updates.
> 
> I've left a few more comments.

Thanks for the review @david-martin. I've addressed some comments and left one question for you in the interest of getting your opinion. 
 
> I know I'm being pedantic, but I really want to minimise the potential for human error.

Your feedback has been valuable. It is absolute fine with me.

david-martin reviewed (approved):
Thanks again @machi1990 

@robshelly Will hold off on merge until you've had a chance to review
